src/core/mantle/evaluator/simple/index.ts

```diff
@@ -20,9 +20,10 @@ export async function evalSimpleLayer(
   ctx: EvalContext,
 ): Promise<EvalResult | undefined> {
   const features = layer.data ? await ctx.getAllFeatures(layer.data) : undefined;
+  const firstFeature = [...(features || [])].shift();
   const appearances: Partial<LayerAppearanceTypes> = pick(layer, appearanceKeys);
   return {
-    layer: evalLayerAppearances(appearances, layer),
+    layer: evalLayerAppearances(appearances, layer, firstFeature),
```

Comment:
```suggestion
    layer: evalLayerAppearances(appearances, laye),
```

This is probably inappropriate. I think the problem lies at this line:

```ts
features?.map(f => evalSimpleLayerFeature(layer, f)),
```

Reply:
Not really, the problem is that we are calling `evalLayerApperances` without any features in it and VariableReplacer doesn't like it because if the used expression is JSONPATH variableReplacer demands features(properties) which wouldn't be provided in the above example. That's the cause of this BUG.

Reply:
But I agree with you on the inappropriate part.